packages/core/src/browser/widgets/widget.ts

```diff
@@ -364,23 +365,37 @@ function waitForVisible(widget: Widget, visible: boolean, attached?: boolean): P
     });
 }
 
+const pinnedTitles = new Map<Title<Widget>, [boolean, Slot<Widget, void>]>();
```

Comment:
```suggestion
const pinnedTitles = new Map<Title<Widget>, PinnedProperties>();
```

Can we maybe create a type for the values of the map? This way the values in the array are better described and the code in the unpin method becomes more readable imho. I am not sure about the name, but something along the lines of `PinnedProperties` could be a good fit.
WDYT?

Reply:
There already is a type for the values in the map, it's a tuple of boolean and a Slot. Explicitly naming that type might make sense if the type was used elsewhere for documentatoin, but it's all private and happening in a couple of lines, so I don't think it adds any value.